softioc/device.py

```diff
@@ -293,6 +293,7 @@ class WaveformBase(ProcessDeviceSupportCore):
     #   NELM    Length of allocated array in number of elements
     #   NORD    Currently reported length of array (0 <= NORD <= NELM)
     _fields_ = ['UDF', 'FTVL', 'BPTR', 'NELM', 'NORD']
+    dtype = None
```

Comment:
Probably worth adding a one line comment above this to explain why!

(Sorry, didn't mention this before; best to force push the comment, please.)

Reply:
Just to prevent lots of forced pushes, is this okay:
```
_fields_ = ...
# Allow set() to be called before init_record():
dtype = None
```

Reply:
I've made the changes as I described above. Easy enough to modify if you're not happy with the comment

Reply:
That's odd... I think the comment line just vanished!

Reply:
Okay, let's just pretend that never happened. I think I've fixed it now...